types/store/plugins.ts

```diff
@@ -38,7 +39,7 @@ export type PostPluginComponent = {
     id: string;
     pluginId: string;
     type: string;
-    component: React.Component;
+    component: React.ComponentType<{post: Post}>;
```

Comment:
When a react component is used like in the following code snippet: 

```javascript
if (pluginPostCardTypes.hasOwnProperty(postType)) {
            const PluginComponent = pluginPostCardTypes[postType].component;
            content = <PluginComponent post={selected}/>;
        }
```

TypeScript is looking for a type not a constructor function, so it was giving an error that the `PluginComponent` variable `does not have any construct or call signatures`. So, had to change it.

Here are couple of links for reference:

https://github.com/microsoft/TypeScript/issues/28631

https://stackoverflow.com/questions/31815633/what-does-the-error-jsx-element-type-does-not-have-any-construct-or-call